ddtrace/contrib/internal/celery/app.py

```diff
@@ -96,3 +105,27 @@ def _traced_beat_inner(func, instance, args, kwargs):
             return func(*args, **kwargs)
 
     return _traced_beat_inner
+
+
+def _traced_apply_async_function(integration_config, fn_name, resource_fn=None):
```

Comment:
It likely won't be apparent to future readers why we're grabbing a span out of seemingly thin air and then closing it here. 

This is a perfect time for a comment explaining WHY we're doing this.  I see a good blurb in the test, but it should be added here as well.  I'd include something along the lines of "apply_async synchronously calls the before/after signals, so we make sure we close at the end"